transfer_nlp/plugins/trainers.py

```diff
@@ -395,16 +411,26 @@ def update_engine(self, engine, batch):
         if engine.state.iteration % self.loss_accumulation_steps == 0:
             self.optimizer.step()
             self.optimizer.zero_grad()
-
-        return self.output_transform(y_pred, batch['y_target'], loss.item())
+        if isinstance(batch, dict):
+            return self.output_transform(y_pred, batch['y_target'], loss.item())
+        elif isinstance(batch, tuple) or isinstance(batch, list):
+            return self.output_transform(y_pred, batch[-1], loss.item())
+        else:
+            raise ValueError("Only dict tuples and lists are valid for batch")
 
     def infer_engine(self, engine, batch):
 
         self.model.eval()
         with torch.no_grad():
             batch = _prepare_batch(batch, device=self.device, non_blocking=False)
-            y_pred = self._forward(batch)
-            return self.eval_output_transform(y_pred, batch['y_target'])
+            if isinstance(batch, dict):
+                y_pred = self._forward(batch)
+                return self.eval_output_transform(y_pred, batch['y_target'])
+            elif isinstance(batch, tuple) or isinstance(batch, list):
+                y_pred = self.model.forward(*batch[:-1])
```

Comment:
is this right? when doing inference you may not have labels.

Reply:
i guess the old dict approach also required y_target...

Reply:
The idea here is to use ignite trainers for training only (which is, ignite only sees labeled data in training and evaluation/test time), and then use model's calls for unlabeled inference